### PR TITLE
Fix unused parameter warnings

### PR DESCRIPTION
**Pull request by Rombur**

### CODE REVIEW

File: packages/Search/test/tstLinearBVH.cpp [hunk 1]
```diff
@@ -515,7 +515,7 @@ std::vector<std::array<double, 3>> make_random_cloud( double Lx, double Ly,
 {
     std::vector<std::array<double, 3>> cloud( n );
     std::default_random_engine generator;
-    std::uniform_real_distribution<double> distribution_x( 0.0, Lz );
+    std::uniform_real_distribution<double> distribution_x( 0.0, Lx );
```
Comment (dalg24) [inline packages/Search/test/tstLinearBVH.cpp]: I thought you got "all" of these...

File: packages/Utils/test/tstMiscellaneous.cpp [hunk 2]
```diff
@@ -16,10 +16,10 @@
 TEUCHOS_UNIT_TEST( DataTransferKitRuntimeAPI, return_version )
 {
     auto const dtk_version = DataTransferKit::version();
-    TEUCHOS_ASSERT( !dtk_version.empty() );
+    TEST_ASSERT( !dtk_version.empty() );
```
Comment (dalg24) [inline packages/Utils/test/tstMiscellaneous.cpp]: I had these changes sitting on some branch :)
Good catch.  Did you actually grep for `TEUCHOS_`?

**Comment by Rombur** [comment]
Don't merge this yet. I will change the `configure` script so Jenkins can catch these warnings next time

File: packages/Search/test/tstLinearBVH.cpp (hunk 1 above)
Comment (Rombur) [reply packages/Search/test/tstLinearBVH.cpp]: me too :(

File: packages/Utils/test/tstMiscellaneous.cpp (hunk 2 above)
Comment (Rombur) [reply packages/Utils/test/tstMiscellaneous.cpp]: No, I got a warning that `out` and `success` were not used.

Comment (dalg24) [reply packages/Utils/test/tstMiscellaneous.cpp]: Please do so and change if there is any other instance.

**Comment by Rombur** [comment]
The problem is not the compiler but the flags that we are passing to gcc. These are not unused variables but unused parameters. We need `-Wextra` to get these warnings.

File: packages/Utils/test/tstMiscellaneous.cpp (hunk 2 above)
Comment (Rombur) [reply packages/Utils/test/tstMiscellaneous.cpp]: I just grepped and there shouldn't be any other `TEUCHOS_ASSERT`

**Comment by Rombur** [comment]
I have fixed more unused parameter, enable shadow warnings, and added `Wextra` but only for clang so that we are not swamped with warnings when using the docker container.

**Comment by Rombur** [comment]
Should be good to merge. We still have about 200 warnings split equally between Kokkos, Tpetra, and Teuchos.

**Pull request merged**